### PR TITLE
Moved SAP adapter from carbon mediation

### PR DESCRIPTION
**Pull request by GDLMadushanka**

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -0,0 +1,121 @@
...
+
+    <artifactId>sap-adapter</artifactId>
+    <version>4.7.33-SNAPSHOT</version>
+    <packaging>pom</packaging>
+
+    <modules>
+        <module>components/sap</module>
+        <module>features/org.wso2.carbon.sap.feature</module>
+    </modules>
+
```
Comment (milindaperera) [inline pom.xml]: Let's move all the dependencies to root pom file

File: components/sap/src/main/java/org/wso2/carbon/transports/sap/CarbonDestinationDataProvider.java [hunk 2]
```diff
@@ -0,0 +1,141 @@
+/*
+ *  Copyright (c) 2005-2008, WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
```
Comment (sameeragunarathne) [inline components/sap/src/main/java/org/wso2/carbon/transports/sap/CarbonDestinationDataProvider.java]: licence headers should be updated

**Comment by milindaperera** [comment]
Let's move the files preserving the commit history

File: pom.xml (hunk 1 above)
Comment (GDLMadushanka) [reply pom.xml]: +1 done

File: components/sap/src/main/java/org/wso2/carbon/transports/sap/CarbonDestinationDataProvider.java (hunk 2 above)
Comment (GDLMadushanka) [reply components/sap/src/main/java/org/wso2/carbon/transports/sap/CarbonDestinationDataProvider.java]: I picked files with a commit history

**Pull request merged**